src/typval.c

```diff
@@ -1760,7 +1759,7 @@ typval_compare_object(
 
     for (int i = 0; i < cl1->class_obj_member_count; ++i)
 	if (!tv_equal((typval_T *)(obj1 + 1) + i,
```

Comment:
The `recursive` should also be FALSE, because it is the first time to call `tv_equal()`